android/app/src/main/java/com/mattermost/share/ShareModule.java

```diff
@@ -145,17 +145,19 @@ public WritableArray processIntent() {
                 items.pushMap(map);
             } else if (Intent.ACTION_SEND.equals(action)) {
                 Uri uri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
-                text = "file://" + RealPathUtil.getRealPathFromURI(currentActivity, uri);
-                map.putString("value", text);
+                if (uri != null) {
+                    text = "file://" + RealPathUtil.getRealPathFromURI(currentActivity, uri);
+                    map.putString("value", text);
 
-                if (type.equals("image/*")) {
-                    type = "image/jpeg";
-                } else if (type.equals("video/*")) {
-                    type = "video/mp4";
-                }
+                    if (type.equals("image/*")) {
+                        type = "image/jpeg";
+                    } else if (type.equals("video/*")) {
+                        type = "video/mp4";
+                    }
 
-                map.putString("type", type);
-                items.pushMap(map);
+                    map.putString("type", type);
+                    items.pushMap(map);
```

Comment:
nit: Probably not related but looks like we're repeating `map.putString` and `items.pushMap` for each condition.  Is it possible to do it at the bottom?